pyvista/plotting/plotting.py

```diff
@@ -4585,6 +4595,61 @@ def export_obj(self, filename):
         exporter.SetRenderWindow(self.ren_win)
         return exporter.Write()
 
+    @property
+    def _datasets(self):
+        """Return a list of all datasets associated with this plotter."""
+        datasets = []
+        for renderer in self.renderers:
+            for actor in renderer.actors.values():
+                mapper = actor.GetMapper()
+
+                # ignore any mappers whose inputs are not datasets
+                if hasattr(mapper, 'GetInputAsDataSet'):
+                    datasets.append(mapper.GetInputAsDataSet())
+
+        return datasets
+
+    def _remove_added_scalars(self):
+        """Remove any scalars added to this plotter's datasets by this plotter.
+
+        Additional point or cell data may be added to be able to correctly plot
+        scalars. This usually occurs in one of the following cases:
+
+        * ``<scalar-name>-real`` for complex data
+        * ``<scalar-name>-normed`` for normalized multi-component arrays.
+        * ``scalars`` when adding scalars within ``add_mesh`` rather than
+          selecting an existing array.
```

Comment:
Thanks to https://github.com/pyvista/pyvista/pull/2818 I came across another case: when supplying opacity through an array, VTK generates `__custom_rgba`, and even `__custom_rgba-real` if the scalars happened to be complex.

I don't know if we want to mention this case too, but it's probably not a case for which we can reactivate other scalars, so we don't have to change the logic because of this. I'm sure there are other weird examples too, so this need not be comprehensive (and doesn't even claim to be that).

I guess there's also the <scalar-name>-<index> case which is not mentioned here either.

Reply:
> it's probably not a case for which we can reactivate other scalars

Sadly, not yet. We could add that logic in, but there isn't a clean way to do this and it's not enough of an issue for me to invest the time to add it in.

Added the note about `__custom_rgba` and the component array in c24f4abe186fc0d0faad0dedc2dbc9f8aa9c8187.